### ISSUE TITLE
Instability when including convective mass flux

### ISSUE THREAD

**Issue by DavidBrust** [DavidBrust/GasDiffusionElectrodes#1]
When including convective transport in the gas phase species flux term, instability occurs. With instability I mean, that the solution changes a lot when refining the mesh, ultimately leading to unreasonable solutions.

When setting the convective velocity to 0 m/s, the problem vanishes.

**Comment by j-fu** [comment]
I could see what is happening. Wouldn't call it instability. Perhaps it is a scaling problem in the Bernoulli argument or something like that.
The interface with different velocities is problematic when it comes to mass diffusion.


However my main problem is that I doubt a bit the model. How is the velocity defined ? I think it is really necessary here to go to Stefan-Maxwell. I think Weng et al have it, too.

A realization in a different context (but with VoronoiFVM.jl...) can be found [here](https://github.com/fafroo/Nonisothermal-dusty-gas-model/)

**Comment by DavidBrust** [comment]
Thank you for your assessments and providing the link to the implementation in the context of the Dusty gas model.

Weng et. al. calculate a mass averaged gas velocity in the porous media from Darcy's law (eq. 14 in Weng). Because of the "step change" of permeabilities in the different domains, I calculated two different gas velocities from Darcy's law "by hand", using parameter values provided in Weng et al and assuming a pressure drop I deemed reasonable.

As far as I understand, Weng et al. implement "effective, mass averaged" diffusion coefficients (eq. 10 in Weng), a simplification, which accoring to R. Taylor and R. Krishna, Multicomponent mass transfer, is valid if only one species is diffusing through otherwise stagnant gas components. (this situation does not apply in the case of Weng et al)

Moving forward, I will implement the Stefan-Maxwell multi-component species transport in a separate notebook (following the procedure of Nonisothermal-dusty-gas-model). This can then serve as a starting point to incorporate the convective flux.
